# 常见的模型评价指标

In [1]:
import numpy as np
import matplotlib.pyplot as plt

## 1、均方误差
MSE(Mean Square Error)

$$MSE=\frac{1}{m}\sum_{i=1}^m(y^{(i)}-\hat{y}^{(i)})^2\tag 1$$

In [2]:
y_true = np.array([1,2,3,4,5])
y_pred = np.array([1.1,2.2,3.1,4.2,5])

In [3]:
def mse(y_true,y_pred):
    #实现公式（1） （1行代码）
    return np.sum(np.power(y_true - y_pred,2)) / len(y_true)

In [4]:
mse(y_true,y_pred)

0.020000000000000035

**预期结果：**0.02

## 2、均方根误差
RMSE(Root Mean Square Error)

$$RMSE=\sqrt{\frac{1}{m}\sum_{i=1}^m(y^{(i)}-\hat{y}^{(i)})^2}\tag 2$$

In [5]:
def rmse(y_true,y_pred):
    #实现公式（2） （1行代码）
    return np.sqrt(np.sum(np.power(y_true - y_pred,2)) / len(y_true))
rmse(y_true,y_pred)

0.14142135623730964

**预期结果：**0.1414

## 3、平均绝对误差
MAE(Mean Absolute Error)

$$MAE=\frac{1}{m}\sum_{i=1}^m\mid y^{(i)}-\hat{y}^{(i)}\mid \tag 3$$

In [6]:
def mae(y_true,y_pred):
    #实现公式（3） （1行代码）
    return np.sum(np.abs(y_true - y_pred)) / len(y_true)
mae(y_true,y_pred)

0.1200000000000001

**预期结果：**0.12